### PR TITLE
[WIP] 1.5.0

### PR DESCRIPTION
**Pull request by daniel-trinh**
- Autoplugin
- Merged with latest typesafe branch
- Merged in clear-cache-on-preferences-changed

### CODE REVIEW

File: src/main/scala/com/typesafe/sbt/SbtScalariform.scala [hunk 1]
```diff
@@ -36,27 +42,43 @@ object SbtScalariform extends Plugin {
...
+
+  def scalariformSettingsWithIt: Seq[Setting[_]] =
+    defaultScalariformSettingsWithIt ++ List(
       compileInputs in (Compile, compile) <<= (compileInputs in (Compile, compile)) dependsOn (format in Compile),
-      compileInputs in (Test, compile) <<= (compileInputs in (Test, compile)) dependsOn (format in Test)
+      compileInputs in (Test, compile) <<= (compileInputs in (Test, compile)) dependsOn (format in Test),
+      compileInputs in (It, compile) <<= (compileInputs in (It, compile)) dependsOn (format in It)
     )
 
   def defaultScalariformSettings: Seq[Setting[_]] =
```
Comment (gregsymons) [inline src/main/scala/com/typesafe/sbt/SbtScalariform.scala:79]: I think you may want to move defaultScalariformSettings into autoImport; everything in the plugin is no longer automatically imported.

File: src/main/scala/com/typesafe/sbt/SbtScalariform.scala [hunk 2]
```diff
@@ -36,27 +42,43 @@ object SbtScalariform extends Plugin {
...
     noConfigScalariformSettings ++ inConfig(Compile)(configScalariformSettings) ++ inConfig(Test)(configScalariformSettings)
 
+  def defaultScalariformSettingsWithIt: Seq[Setting[_]] =
+    defaultScalariformSettings ++ inConfig(It)(configScalariformSettings)
+
```
Comment (gregsymons) [inline src/main/scala/com/typesafe/sbt/SbtScalariform.scala:84]: Same as defaultScalariformSettings, you probably want to move this into autoImport.

**Comment by dwijnand** [comment]
Do you want have a go and see if adding requires = JvmPlugin helps?

It's sometimes the cause, that we're thinking if perhaps we should change the default: https://github.com/sbt/sbt/issues/2082.

**Comment by daniel-trinh** [comment]
@dwijnand Adding requires = JvmPlugin fixed it. Thanks!

**Pull request merged**